### PR TITLE
Strict type `Dependabot::Gradle::FileFetcher`

### PR DESCRIPTION
**Pull request by raj-meka**
### What are you trying to accomplish?
Sorbet Implementation for Gradle FileFetcher ruby file.

### Anything you want to highlight for special attention from reviewers?
Please pay attention to the fixes in the Gradle FileFetcher ruby file.

### How will you know you've accomplished your goal?
The sorbet rules will fire as needed.

### Checklist

- [x]  I have run the complete test suite to ensure all tests and linters pass.
- [x]  I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x]  I have written clear and descriptive commit messages.
- [x]  I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x]  I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: gradle/lib/dependabot/gradle/file_fetcher.rb [hunk 1]
```diff
@@ -82,24 +87,28 @@ def subproject_buildfiles(root_dir)
           .subproject_paths
 
         subproject_paths.filter_map do |path|
+          @buildfile_name = T.let(@buildfile_name, T.nilable(String))
           if @buildfile_name
-            fetch_file_from_host(File.join(root_dir, path, @buildfile_name))
+            buildfile_path = T.let(File.join(root_dir, path, @buildfile_name), String)
```
Comment (JamieMagee) [inline gradle/lib/dependabot/gradle/file_fetcher.rb:92]: Why is `T.let` necessary here?

File: gradle/lib/dependabot/gradle/file_fetcher.rb [hunk 2]
```diff
@@ -82,24 +87,28 @@ def subproject_buildfiles(root_dir)
...
+            fetch_file_from_host(buildfile_path)
           else
             buildfile(File.join(root_dir, path))
           end
-        rescue Dependabot::DependencyFileNotFound
-          # Gradle itself doesn't worry about missing subprojects, so we don't
-          nil
+      rescue Dependabot::DependencyFileNotFound
+        # Gradle itself doesn't worry about missing subprojects, so we don't
+        nil
```
Comment (JamieMagee) [inline gradle/lib/dependabot/gradle/file_fetcher.rb:99]: It looks like the indentation was decreased a little here.

File: gradle/lib/dependabot/gradle/file_fetcher.rb [hunk 3]
```diff
@@ -82,24 +87,28 @@ def subproject_buildfiles(root_dir)
           .subproject_paths
 
         subproject_paths.filter_map do |path|
+          @buildfile_name = T.let(@buildfile_name, T.nilable(String))
```
Comment (JamieMagee) [inline gradle/lib/dependabot/gradle/file_fetcher.rb:90]: This seems a little odd. Can you create a constructor and create it there?

Comment (GarryHurleyJr) [reply gradle/lib/dependabot/gradle/file_fetcher.rb:90]: Please run it through **rubocop -A** to see if this autocorrects.  whitespace doesn't seem like a big deal, but that rule should just check for issues which have already been corrected and make the code less likely to have issues when someone else rebases.

File: gradle/lib/dependabot/gradle/file_fetcher.rb (hunk 2 above)
Comment (GarryHurleyJr) [reply gradle/lib/dependabot/gradle/file_fetcher.rb:99]: @raj-meka Please run it through **rubocop -A** to see if this autocorrects.  whitespace doesn't seem like a big deal, but that rule should just check for issues which have already been corrected and make the code less likely to have issues when someone else rebases.

File: gradle/lib/dependabot/gradle/file_fetcher.rb (hunk 1 above)
Comment (raj-meka) [reply gradle/lib/dependabot/gradle/file_fetcher.rb]: removed it

File: gradle/lib/dependabot/gradle/file_fetcher.rb (hunk 2 above)
Comment (JamieMagee) [reply gradle/lib/dependabot/gradle/file_fetcher.rb:99]: @raj-meka it doesn't look like you resolved this

**Comment by JamieMagee** [comment]
@raj-meka please don't mark PR comments as resolved if you don't address them. If you disagree, at least leave a comment explaining why.

**Review by JamieMagee — APPROVED** [review]

**Review by sachin-sandhu — APPROVED** [review]
Approving changes.

**Pull request merged**